src/devdocs2zim/generator.py

```diff
@@ -491,6 +514,34 @@ def generate_zim(
             )
         return zim_path
 
+    @staticmethod
+    def fetch_logo_bytes(user_logo_path: str) -> bytes:
+        """Fetch a user-supplied logo for the ZIM and format/resize it.
+
+        Parameters:
+          user_logo_path: Path or URL to the logo.
+        """
+        logger.info(f"  Fetching logo from: {user_logo_path}")
+        full_logo_path = handle_user_provided_file(source=user_logo_path)
+        if full_logo_path is None:
+            # This appears to only happen if the path is blank.
+            raise Exception(f"Fetching logo {user_logo_path!r} failed.")
+
+        converted_buf = io.BytesIO()
+        if format_for(full_logo_path, from_suffix=False) == "SVG":
+            # SVG conversion generates a PNG in the correct size
+            # so immediately return it.
+            convert_svg2png(full_logo_path, converted_buf, 48, 48)
+            return converted_buf.getvalue()
+        else:
+            # Convert to PNG
+            convert_image(full_logo_path, converted_buf, fmt="PNG")
+
+        # resize to 48x48
```

Comment:
One more thing, shouldn't we do the resize "in_place" by not setting a `dst` param?  no need to keep both the original and the resized in memory (even if it will in fact, but lower level, less code / variables here is always better IMHO). And you should probably indent this resizing code in the `else` branch so this is a bit less confusing.

Reply:
Ah, good call. Fixed!